### PR TITLE
Migrate 4 APIs about array to paddle.tensor.*

### PR DESCRIPTION
**Pull request by liym27**
### PR types

Others
### PR changes

APIs
### Describe

4 APIs: array_length, array_read, array_write, create_array

---

- array_length
image

- array_read
image

- array_write
image

- create_array
image

---

- [x] 修改动转静代码中的接口调用路径
- [ ] 中文文档

### CODE REVIEW

File: python/paddle/tensor/array.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            print(arr_len)  # 1
+    """
+    return layers.array_length(array)
+
+
+def array_read(array, i):
+    """
+    This OP is used to read data at the specified position from the input array.
+
+    Case 1:
```
Comment (zhhsplendid) [inline python/paddle/tensor/array.py]: There is Case 1 but not other cases?

File: python/paddle/tensor/array.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+
+def array_write(x, i, array=None):
+    """
+    This OP writes the input ``x`` into the i-th position of the ``array`` returns the modified array.
+    If ``array`` is none, a new array will be created and returned.
+
+    Args:
+        x (Tensor): The input data to be written into array. It's multi-dimensional
+            Tensor or LoDTensor. Data type: float32, float64, int32, int64.
```
Comment (zhhsplendid) [inline python/paddle/tensor/array.py]: Does it support bool?

File: python/paddle/tensor/array.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    return layers.array_write(x, i, array)
+
+
+def create_array(dtype):
+    """
+    This OP creates an array. It is used as the input of :ref:`api_paddle_tensor_array_read` and
+    :ref:`api_paddle_tensor_array_write`.
+
+    Args:
+        dtype (str): The data type of the elements in the array. Support data type: float32, float64, int32, int64.
```
Comment (zhhsplendid) [inline python/paddle/tensor/array.py]: Same as above comment: does it support bool?

File: python/paddle/tensor/array.py (hunk 2 above)
Comment (liym27) [reply python/paddle/tensor/array.py]: Yes. `bool` has been added.

File: python/paddle/tensor/array.py (hunk 3 above)
Comment (liym27) [reply python/paddle/tensor/array.py]: Yes. `bool` has been added.

File: python/paddle/tensor/array.py (hunk 1 above)
Comment (liym27) [reply python/paddle/tensor/array.py]: Yes. Use `Case` replace `Case 1`

**Review by zhhsplendid — DISMISSED** [review]
LGTM

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

File: python/paddle/tensor/array.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+            item = paddle.tensor.array_read(arr, i)
+            print(item)     # [[5., 5., 5.]]
+    """
+    return layers.array_write(x, i, array)
+
+
+def create_array(dtype):
+    """
+    This OP creates an array. It is used as the input of :ref:`api_paddle_tensor_array_read` and
```
Comment (TCChenlong) [inline python/paddle/tensor/array.py]: 这里label应该是错了，所以预览的时候有问题，应该是 api_paddle_tensor_array_array_read 和 api_paddle_tensor_array_array_write

**Review by lanxianghit — DISMISSED** [review]

File: python/paddle/tensor/array.py (hunk 4 above)
Comment (liym27) [reply python/paddle/tensor/array.py]: 好的，感谢建议，已修改

**Review by TCChenlong — APPROVED** [review]
LGTM

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**